core/os/time.cpp

```diff
@@ -293,7 +297,7 @@ int64_t Time::get_unix_time_from_datetime_dict(const Dictionary p_datetime) cons
 }
 
 int64_t Time::get_unix_time_from_datetime_string(String p_datetime) const {
-	PARSE_ISO8601_STRING
+	PARSE_ISO8601_STRING(-1)
 	VALIDATE_YMDHMS(0)
```

Comment:
With this code:
* If the string parsing fails due to an incorrect number of elements, it will return -1
* If one of the values in each element is invalid, it will return 0

It may be useful to have different return values for these cases. I'm not sure which behavior is desired, but I think this is good.